server.js

```diff
@@ -0,0 +1,79 @@
+var http = require("http");
+var secondLargest = require("./programs/secondLargest").secondLargest;
+var calculateFrequency = require("./programs/calculateFrequency")
+  .calculateFrequency;
+var flatten = require("./programs/flatten").flatten;
+var unflatten = require("./programs/unflatten").unflatten;
+const util = require("util");
+
+var input_array = [
+  21,
+  87,
+  91,
+  31,
+  97,
+  59,
+  8,
+  83,
+  27,
+  23,
+  76,
+  72,
+  31,
+  91,
+  65,
+  77,
+  84,
+  27,
+  56,
+  59
+];
+var input_string = "shali is running fast";
+var unflatObject = {
+  flatJSON: false,
+  i: { am: { not: { so: { flat: true, unflat: false } }, a: "tree" } },
+  dates: [{ day: 1 }, { day: 8947 }]
+};
+var flatObject = {
+  flatJSON: false,
+  "i.am.not.so.flat": true,
+  "i.am.not.so.unflat": false,
+  "i.am.a": "tree",
+  "dates.0.day": 1,
+  "dates.1.day": 8947
+};
+
+http
+  .createServer(function(req, res) {
+    res.writeHead(200, { "Content-Type": "text/html" });
+    res.write(
```

Comment:
If you want to pretty print the output, then the logic should be with the library providing that output, abstracted from the one calling it. If the caller just needs the output he can get it, if wants a pretty printed version he can get it.

Reply:
I will remember it for next time. In here if i make it so it will require extra functions for pretty print so i am made it as default as this program call all 4 functions and the input is also part of program and not a taken from user